tests/tokenizer-tests.ts

```diff
@@ -217,6 +222,38 @@ QUnit.test('A newline immediately following a <textarea> tag is stripped', funct
   assert.deepEqual(tokens, [startTag('textarea'), chars('hello'), endTag('textarea')]);
 });
 
+// https://html.spec.whatwg.org/multipage/semantics.html#the-title-element
+QUnit.test('The title element content is always text', function(assert) {
+  let tokens = tokenize("<title>&quot;hey <b>there</b><!-- comment --></title>");
+  assert.deepEqual(tokens, [startTag('title'), chars('"hey <b>there</b><!-- comment -->'), endTag('title')]);
+});
+
+// https://html.spec.whatwg.org/multipage/semantics.html#the-style-element
+QUnit.test('The style element content is always text', function(assert) {
+  let tokens = tokenize("<style>&quot;hey <b>there</b><!-- comment --></style>");
+  assert.deepEqual(tokens, [startTag('style'), chars('"hey <b>there</b><!-- comment -->'), endTag('style')]);
+});
+
+// https://html.spec.whatwg.org/multipage/scripting.html#restrictions-for-contents-of-script-elements
+QUnit.test('The script element content restrictions', function(assert) {
+  let tokens = tokenize("<script>&quot;hey <b>there</b><!-- comment --></script>");
+  assert.deepEqual(tokens, [startTag('script'), chars('"hey <b>there</b><!-- comment -->'), endTag('script')]);
```

Comment:
> chars('"hey <b>there</b><!-- comment -->')

It is not clear to me that we actually want the `&quot;` to be transformed into `"` (for `script` and `style` that is), I _think_ we should pass it through without transforming the char refs. Have I misread the spec?

Reply:
Ah, the copy&paste TDD! 😅 I took the `<title>` test, and used it as a base for script and style tests, and then the implementation… Working on a fix. 😉